src/execution/lazy.rs

```diff
@@ -354,24 +348,20 @@ impl ast::Scan {
                 lazy_graph: exec.lazy_graph,
                 function_parameters: exec.function_parameters,
                 prev_element_debug_info: exec.prev_element_debug_info,
+                error_context: exec.error_context.clone(),
                 shorthands: exec.shorthands,
                 cancellation_flag: exec.cancellation_flag,
             };
 
             for statement in &arm.statements {
-                arm_exec
-                    .cancellation_flag
-                    .check("executing scan statements")?;
+                arm_exec.error_context.statement = format!("{}", statement);
+                arm_exec.error_context.statement_location = statement.location();
```

Comment:
The `StatementContext` is updated in nested blocks, so that it always represents the innermost statement.